BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,341 @@
+            aGameRecord[kSpatialSpanMemoryGameRecordTargetRectsKey] = rectangles
+            
+            gameRecords += [aGameRecord]
+        }
+        memoryGameResults[kSpatialSpanMemorySummaryGameRecordsKey] = gameRecords
+        let filename = bridgifyFilename(self.identifier) + ".json"
+        memoryGameResults[kItemKey] = filename
+        return ArchiveableResult.init(result: (memoryGameResults as NSDictionary).jsonObject(), resultType: .Dictionary, filename: filename)
+    }
+    
+    
+    func makeTouchSampleRecords(touchSamples: [ORKSpatialSpanMemoryGameTouchSample]?) -> NSArray
+    {
+        var samples = [[String: AnyObject]]()
+        
+        guard touchSamples != nil else {
+            return samples
+        }
+        for sample: ORKSpatialSpanMemoryGameTouchSample in touchSamples! {
+            
+            var aTouchSample = [String: AnyObject]()
+            
+            aTouchSample[kSpatialSpanMemoryTouchSampleTimeStampKey]   = sample.timestamp
+            aTouchSample[kSpatialSpanMemoryTouchSampleTargetIndexKey] = sample.targetIndex
+            aTouchSample[kSpatialSpanMemoryTouchSampleLocationKey]    = NSStringFromCGPoint(sample.location)
+            aTouchSample[kSpatialSpanMemoryTouchSampleIsCorrectKey]   = sample.correct
+            
+            samples += [aTouchSample]
+        }
+        return  samples
+    }
+    
+    func makeTargetRectangleRecords(targetRectangles: [NSValue]) -> NSArray
+    {
+        var rectangles = [String]()
+        
+        for value: NSValue in targetRectangles {
+            let rectangle = value.CGRectValue()
+            let stringified = NSStringFromCGRect(rectangle)
+            rectangles += [stringified]
+        }
+        return  rectangles
+    }
+}
+
+class AnswerKeyAndValue: NSObject {
+    var key: String
+    var value: AnyObject?
+    
+    init(key: String, value: AnyObject?) {
+        self.key = key
+        self.value = value
+    }
+}
+
+protocol ORKQuestionResultAnswerJSON {
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)?
+}
+
+private var _timeOnlyFormatter: NSDateFormatter? = nil
+
+extension ORKQuestionResult: ORKQuestionResultAnswerJSON {
+    
+    func timeOnlyFormatter() -> NSDateFormatter {
+        if _timeOnlyFormatter == nil {
+            _timeOnlyFormatter = NSDateFormatter.init()
+            _timeOnlyFormatter!.dateFormat = "HH:mm:ss.SSS"
+            let enUSPOSIXLocale = NSLocale.init(localeIdentifier: "en_US_POSIX");
+            _timeOnlyFormatter!.locale = enUSPOSIXLocale
+        }
+        
+        return _timeOnlyFormatter!
+    }
+    override func bridgeData(stepIdentifier: String) -> ArchiveableResult? {
+        let filename = bridgifyFilename(self.identifier) + ".json"
+        return ArchiveableResult.init(result: self.jsonObject(), resultType: .Dictionary, filename: filename)
+    }
+    
+    public override func jsonObject() -> AnyObject {
```

Comment:
yes it builds, 'override' is because I didn't remove that qualifier after discovering that didn't actually work in Swift (will do so now), and jsonObject() is the right method name, I think, because it returns a JSON-safe dictionary, not just a dictionary. It can't override jsonObject() implementations in superclasses, but it can still implement the SBAJSONObject protocol itself.